tests/e2e/tests/content-manager/history.spec.ts

```diff
@@ -281,6 +280,35 @@ describeOnCondition(edition === 'EE')('History', () => {
       await expect(page.getByText('titleRename')).toBeVisible();
       await page.getByRole('status').getByText('New field');
     });
+
+    test('A user should be able to restore a history version', async ({ page }) => {
+      await page.getByRole('link', { name: 'Content Manager' }).click();
+      await page.getByRole('link', { name: /Create new entry/, exact: true }).click();
+      await page.waitForURL(ARTICLE_CREATE_URL);
+
+      const titleInput = await page.getByRole('textbox', { name: 'title' });
+      // Create an initial entry to also create an initial version
+      await titleInput.fill('Being from Kansas');
+      await page.getByRole('button', { name: 'Save' }).click();
+      await page.waitForURL(ARTICLE_EDIT_URL);
+      // Update to create another version
+      await titleInput.fill('Being from Florida');
+      await page.getByRole('button', { name: 'Publish' }).click();
+
+      await goToHistoryPage(page);
```

Comment:
Based on my experience with the flaky tests, I recommend awaiting tons of stuff to be extra safe. Here I think you can use the function to check that the success notification has appeared before opening history.

You may get an error because there could be two success notifications at once, after the create action. To prevent it you could replace the `await page.waitForURL(ARTICLE_EDIT_URL)` above by `closeSuccessNotification` so that it gets dismissed